jskit/encoders/bi.py

```diff
@@ -101,16 +101,48 @@ def cosine_sim(vec_a: List[float], vec_b: List[float], meta):
 
 
 @jaseci_action(act_group=['bi_enc'], allow_remote=True)
-def infer(contexts: List, candidates: List):
+def infer(contexts: Union[List[str], List[List]], candidates: Union[List[str], List[List]], context_type: str, candidate_type: str):
     """
     Take list of context, candidate and return nearest candidate to the context
     """
     global model
     model.eval()
-    predicted_candidate = get_inference(model, tokenizer,
-                                        contexts=contexts,
-                                        candidates=candidates)
-    return predicted_candidate
+    predicted_candidates = []
+    if (context_type == "text") and (candidate_type == "text"):
+        predicted_candidates = get_inference(model, tokenizer,
+                                             contexts=contexts,
+                                             candidates=candidates)
+    elif (context_type == "text") and (candidate_type == "embedding"):
+        con_embed = []
+        con_embed = get_context_emb(contexts)
+        for data in con_embed:
+            score_dat = []
+            for lbl in candidates:
+                score_dat.append(cosine_sim(
+                    vec_a=data, vec_b=lbl, meta="string"))
+            predicted_candidates.append(float(np.argmax(score_dat)))
+
+    elif (context_type == "embedding") and (candidate_type == "text"):
+        cand_embed = []
+        predicted_candidates = []
+        cand_embed = get_candidate_emb(candidates)
+        for data in contexts:
+            score_dat = []
+            for lbl in cand_embed:
+                score_dat.append(cosine_sim(
+                    vec_a=data, vec_b=lbl, meta="string"))
+            predicted_candidates.append(candidates[np.argmax(score_dat)])
+    elif (context_type == "embedding") and (candidate_type == "embedding"):
+        for data in contexts:
+            score_dat = []
+            for lbl in candidates:
+                score_dat.append(cosine_sim(
+                    vec_a=data, vec_b=lbl, meta="string"))
+            predicted_candidates.append(float(np.argmax(score_dat)))
+    else:
+        raise HTTPException(status_code=404, detail=str(
+            "input type not supported"))
```

Comment:
Let's make this error message more informative. Let's also provide what input type are supported and what are being passed in actually.